src/io_functions.f90

```diff
@@ -0,0 +1,42 @@
+module io_functions
+    !! This module contains procedures that support IO operations
+
+    use uEMEP_definitions, only: unit_logfile
+
+    implicit none
+    private
+
+    public :: check_dir_exist
+
+contains
+
+    function check_dir_exist(path) result(exists)
+        !! Checks if a directory exists and/or if we have write permission
+        !! in that directory.
```

Comment:
If I read this correctly, it should be `check_dir_writable` because pure existence will fail. (and remove `or` from the docs.)
Or provide an optional parameter to switch between pure existence and writability.

Reply:
Yes, check_dir_writable is maybe a better name. However, it does actually do both. The function will return false if either 1) the directory does not exist or 2) if it exists but is not writable. So checking for 'pure existence' will behave the same as 'writability'. 

The error code could be used to distinguish between the two, but since they are different between ifort and gfortran, it only checks for /= 0.